lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -561,12 +566,73 @@ void OpenCLFunction::execute() {
 
       if (auto *SI = dyn_cast<SplatInst>(&I)) {
         // Pass the splat as a parameter.
-        setKernelArg(kernel, numArgs + 1, SI->getValue());
+        setKernelArg(kernel, ++numArgs, SI->getValue());
       } else if (auto *EPI = dyn_cast<ElementPowInst>(&I)) {
         // Pass the exp as a parameter.
-        setKernelArg(kernel, numArgs + 1, EPI->getExp());
+        setKernelArg(kernel, ++numArgs, EPI->getExp());
       }
 
+      if (isQuantized) {
+        if (isa<ElementAddInst>(I) || isa<ElementSubInst>(I) ||
+            isa<ElementMulInst>(I) || isa<ElementDivInst>(I) ||
+            isa<ElementMinInst>(I) || isa<ElementMaxInst>(I)) {
+          int32_t destOffset = I.getOperand(0).first->getType()->getOffset();
+          float destScale = I.getOperand(0).first->getType()->getScale();
+          auto lhsScaleParams = quantization::quantizeScaleOffset32To8(
+              I.getOperand(1).first->getType()->getScale() / destScale,
+              I.getOperand(1).first->getType()->getOffset());
+          auto rhsScaleParams = quantization::quantizeScaleOffset32To8(
+              I.getOperand(2).first->getType()->getScale() / destScale,
+              I.getOperand(2).first->getType()->getOffset());
+          setKernelArg(kernel, ++numArgs, destOffset);
+          if (isa<ElementMulInst>(I) || isa<ElementDivInst>(I)) {
+            auto resultScaleParams = quantization::quantizeScaleOffset32To8(
+                I.getOperand(1).first->getType()->getScale() *
+                    I.getOperand(2).first->getType()->getScale() / destScale,
+                0);
+            setKernelArg(kernel, ++numArgs, lhsScaleParams.offset_);
```

Comment:
you could pass rhsScaleParams and lhsScaleParams in both cases and remove else condition, might look cleaner.